pkg/deploy/api/types.go

```diff
@@ -328,3 +328,22 @@ type DeploymentConfigRollbackSpec struct {
 	// IncludeStrategy specifies whether to include the deployment Strategy.
 	IncludeStrategy bool
 }
+
+// DeploymentLogs represents the logs for a deployment
+type DeploymentLogs struct {
+	kapi.TypeMeta
+	kapi.ListMeta
+}
+
+// DeploymentLogOptions is the REST options for a deployment log
+type DeploymentLogOptions struct {
+	kapi.TypeMeta
+
+	// Follow if true indicates that the deployment log should be streamed until
+	// the deployment terminates.
+	Follow bool
+
+	// NoWait if true causes the call to return immediately even if the deployment
+	// is not available yet. Otherwise the server will wait until the deployment has started.
+	NoWait bool
```

Comment:
Cesar added this in BuildLogOptions. I am more worried that this should be upstream. @csrwng does it make sense to have this as a field in PodLogOptions and instead of checking for the build/deployment phase && this field, check for pod phase && this field [here](https://github.com/GoogleCloudPlatform/kubernetes/blob/c25bcaa395e4e80ca220af6d0e28f660e69a10a7/pkg/registry/pod/rest.go#L196)?